File: back/mallook/src/main/java/io/ssafy/mallook/domain/product/entity/Product.java
package io.ssafy.mallook.domain.product.entity;

import io.ssafy.mallook.domain.BaseEntity;
import io.ssafy.mallook.domain.shoppingmall.entity.ShoppingMall;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name="product")
public class Product extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @ManyToOne
    @JoinColumn(name="shoppingmall_id")
    private ShoppingMall shopingmall;

    @NotNull
    @Enumerated(EnumType.STRING)
    private MainCategory mainCategory;

    @NotNull
    @Enumerated(EnumType.STRING)
    private SubCategory subCategory;

    @NotBlank
    private String name;

    @NotNull
    private Integer price;

    @NotNull
    private Integer quantity;

    private String brandName;

    @NotBlank
    private String size;

    @NotBlank
    private String color;

    @NotNull
    private Integer fee;

    private String image;

    private String code;

    private String url;

    @ElementCollection
    private List<String> keywordList;
}

File: app/mallook/lib/feature/home/models/product.dart
class Product {
  final String? mainCategory;
  final String? subCategory;
  final String name;
  final int price;
  final int? quantity;
  final String? brandName;
  final String? size;
  final int? fee;
  final String? image;
  final String? code;
  final String? url;

  Product({
    this.mainCategory,
    this.subCategory,
    required this.name,
    required this.price,
    this.quantity,
    this.brandName,
    this.size,
    this.fee,
    this.image,
    this.code,
    this.url,
  });

  Product.fromJson(dynamic json)
      : mainCategory = json['mainCategory'],
        subCategory = json['subCategory'],
        name = json['name'] ?? '',
        price = json['price'] ?? 0,
        quantity = json['quantity'],
        brandName = json['brandName'],
        size = json['size'],
        fee = json['fee'],
        image = json['image'],
        code = json['code'],
        url = json['url'];

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['mainCategory'] = mainCategory;
    map['subCategory'] = subCategory;
    map['name'] = name;
    map['price'] = price;
    map['quantity'] = quantity;
    map['brandName'] = brandName;
    map['size'] = size;
    map['fee'] = fee;
    map['image'] = image;
    map['code'] = code;
    map['url'] = url;
    return map;
  }
}
